lib/matplotlib/colors.py

```diff
@@ -958,6 +958,743 @@ def scaled(self):
+        return f
+
+    def _build_finv(self):
+        vmin = self.vmin
+        vmax = self.vmax
+        self._refpoints_data[0] = vmin
+        self._refpoints_data[-1] = vmax
+        rp_d = self._refpoints_data
+        rp_cm = self._refpoints_cm
+        widths_cm = np.diff(rp_cm)
+        widths_d = np.diff(rp_d)
+
+        masks = []
+        funcs = []
+
+        for i in range(len(widths_cm)):
+            if i == 0:
+                mask = (lambda x, i=i: (x >= rp_cm[i]) * (x <= rp_cm[i + 1]))
+            else:
+                mask = (lambda x, i=i: (x > rp_cm[i]) * (x <= rp_cm[i + 1]))
+            masks.append(mask)
+            funcs.append(lambda x, i=i: self._finvlist[i](
+                (x - rp_cm[i]) / widths_cm[i]) * widths_d[i] + rp_d[i])
+
+        maskmaker = (lambda x: [np.array([mi(x)]) if np.isscalar(
+            x) else mi(x) for mi in masks])
+        finv = (lambda x: np.piecewise(x, maskmaker(x), funcs))
+        return finv
+
+    def _update_f(self, vmin, vmax):
+        self._f = self._build_f()
+        self._finv = self._build_finv()
+        return
+
+    def ticks(self, nticks=None):
+        """
+        Returns an automatic list of `nticks` points in the data space
+        to be used as ticks in the colorbar.
+
+        Parameters
+        ----------
+        nticks : integer, optional
+            Number of ticks to be returned. Default 13.
+
+        Returns
+        -------
+        ticks : ndarray
+            1d array of length `nticks` with the proposed tick locations.
+
+        """
+        rp_cm = self._refpoints_cm
+        widths_cm = np.diff(rp_cm)
+
+        if nticks is None:
+            nticks = max([13, len(rp_cm)])
+
+        if nticks < len(rp_cm):
+            ValueError(
+                "the number of ticks must me larger "
+                "that the number or intervals +1")
+
+        ticks = rp_cm.copy()
+
+        available_ticks = nticks - len(-rp_cm)
+        distribution = widths_cm * (available_ticks) / widths_cm.sum()
+        nticks_each = np.floor(distribution)
+
+        while(nticks_each.sum() < available_ticks):
+            ind = np.argmax((distribution - nticks_each))
+            nticks_each[ind] += 1
+
+        for i in range(len(nticks_each)):
+            if nticks_each[i] > 0:
+                nticks_this = nticks_each[i]
+                auxticks = np.linspace(rp_cm[i], rp_cm[i + 1], nticks_this + 2)
+                ticks = np.concatenate([ticks, auxticks[1:-1]])
+
+        finalticks = np.zeros(ticks.shape, dtype=np.bool)
+        finalticks[0:len(rp_cm)] = True
+
+        inds = np.argsort(ticks)
+        ticks = ticks[inds]
+        finalticks = finalticks[inds]
+
+        ticks = PiecewiseNorm._round_ticks(self.inverse(ticks), finalticks)
+
+        return ticks
+
+
+class MirrorPiecewiseNorm(PiecewiseNorm):
+    """
+    Normalization alowing a dual `PiecewiseNorm` simmetrically around a point.
```

Comment:
There are types in that docstrings:

alowing -> allowing
simmetrical -> symmetrical.